cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -304,7 +305,40 @@ impl Environment for ManagedEnvironment {
     }
 
     fn activation_path(&mut self, flox: &Flox) -> Result<PathBuf, EnvironmentError2> {
-        self.build(flox)?;
+        let pointer_lock_modified_at = 'time: {
+            let Ok(metadate) = self
+            .path
+            .join(GENERATION_LOCK_FILENAME)
+            .metadata()
```

Comment:
nonblocking suggestion: I don't love using `mtime` of the generation lock as an indicator of whether we need to rebuild. `mtime` seems a bit unreliable: if e.g., I have two branches with different copies of the file, if I do `git checkout branch1 && flox activate && git checkout branch2 && git checkout branch1 && flox activate`, we'll do a second rebuild. I feel it's not unlikely there's other edge cases like that. What do you think of adding the lockfile to the built environment and diffing it? LIke https://github.com/flox/flox/commit/1ea3646ad1ca4212d86ddd0946b973760a839e98

Reply:
Yeah it's not the most precise I agree.

* `git checkout branch1 && flox activate && git checkout branch2 && git checkout branch1 && flox activate` will actually not do a full rebuild, but just evaluate and figure we already got the store path. evaluation takes its time occasionally, but it shouldnt be very frequent(?) and still quicker than the first time we made the env...

* adding the lockfile to the built environment.

I did think of that too and I agree it's probably caching more, maybe too much..)
1. we do not invalidate if the outpath changes due to changes in `buildenv` -- the current solution does neither, but you can always force a new eval/build with `flox activate -- touch '$FLOX_ENV'` 
2. I'd like to keep the "API" between built environments and flox as small as possible.
   It's getting harder to reason about the contents of envs the more "single usecase" files we put in there (also sth about compatibility ...)
3. I also wanted to avoid adding another place to read the lockfile from -- single source of thruth and such..

I think we should be able to switch approaches here in the future if the git switcheroo and early invalidation causes issues for users.